src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -566,7 +566,7 @@ private void ProduceTagsSynchronously(TaggerContext<TTag> context)
                     {
                         await this.ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
                         UpdateStateAndReportChanges(newTagTrees, bufferToChanges, newState, initialTags);
-                    }).CompletesAsyncOperation(asyncToken);
+                    }, cancellationToken).CompletesAsyncOperation(asyncToken);
```

Comment:
this one is scary.  but i'm not sure if it's right/wrong.  both the before and after could be incorrect here.  can you use CancellationToken.None, with a comment saying that that could very well be wrong.

I'm very worried about potential tagging heisenbugs here.